File: src/main/java/com/kingdom/service/LobbyChats.java
package com.kingdom.service;

import com.kingdom.model.LobbyChat;
import com.kingdom.model.User;

import java.util.Date;
import java.util.LinkedList;

public class LobbyChats {

    public static final int MAX_LOBBY_CHATS = 50;

    private static LobbyChats ourInstance = new LobbyChats();

    private LinkedList<LobbyChat> chats = new LinkedList<LobbyChat>();

    public static LobbyChats getInstance() {
        return ourInstance;
    }

    public LinkedList<LobbyChat> getChats() {
        return chats;
    }

    public void addChat(User user, String message) {
        if (chats.size() == MAX_LOBBY_CHATS) {
            chats.removeLast();
        }
        LobbyChat chat = new LobbyChat();
        chat.setUsername(user.getUsername());
        chat.setMessage(message);
        chat.setTime(new Date());
        chats.addFirst(chat);
    }

    public void addPrivateChat(User sender, User receiver, String message) {
        if (chats.size() == MAX_LOBBY_CHATS) {
            chats.removeLast();
        }
        LobbyChat chat = new LobbyChat();
        chat.setUsername(sender.getUsername());
        chat.setMessage(message);
        chat.setTime(new Date());
        chat.setUserId(receiver.getUserId());
        chats.addFirst(chat);
    }
}

File: src/main/java/com/kingdom/service/LobbyChats.kt
package com.kingdom.service

import com.kingdom.model.LobbyChat
import com.kingdom.model.User

import java.util.Date
import java.util.LinkedList

class LobbyChats {

    val chats = LinkedList<LobbyChat>()

    fun addChat(user: User, message: String) {
        if (chats.size == MAX_LOBBY_CHATS) {
            chats.removeLast()
        }
        val chat = LobbyChat(user.username, message, Date())
        chats.addFirst(chat)
    }

    fun addPrivateChat(sender: User, receiver: User, message: String) {
        if (chats.size == MAX_LOBBY_CHATS) {
            chats.removeLast()
        }
        val chat = LobbyChat(sender.username, message, Date())
        chat.userId = receiver.userId
        chats.addFirst(chat)
    }

    companion object {

        const val MAX_LOBBY_CHATS = 50

        val instance = LobbyChats()
    }
}
